docs/sts/web-identity.go

```diff
@@ -90,7 +90,7 @@ func parseDiscoveryDoc(ustr string) (DiscoveryDoc, error) {
 	}
 	defer resp.Body.Close()
 	if resp.StatusCode != http.StatusOK {
-		return d, err
+		return d, errors.New(resp.Status)
```

Comment:
Suggestion, this way actual error is lost. Should we create final error with response and error both before return?

Reply:
here err always nil @shtripat